dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassAndJar.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.opensource.classpath;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.nio.file.Path;
+import java.util.Objects;
+
+/**
+ * A tuple of a class name and a jar file to uniquely locate the class implementation in a class
+ * path.
+ */
+final class ClassAndJar {
+  private final Path jar;
+  private final String className;
+
+  ClassAndJar(Path jar, String className) {
+    this.jar = checkNotNull(jar);
+    this.className = checkNotNull(className);
+  }
+
+  /** Returns jar file containing the class. */
```

Comment:
`Returns the path to the JAR file containing the class.`

Reply:
Updating that: https://github.com/GoogleCloudPlatform/cloud-opensource-java/pull/598/files#diff-dcec2e6026c85f37b7538c5c3313b146R38